tools/edit_instance_template/edit_instance_template.go

```diff
@@ -16,152 +17,285 @@ import (
 		Project:                  *project,
-		InstanceTemplateResource: resp,
+		InstanceTemplateResource: new,
+	}
+	if !*apply {
+		// Don't print the template here, it's fucken huge.
+		log.Printf("Would create new template: %q", newTemplateName)
+		return nil, nil
 	}
-	_, err = c.Insert(ctx, createReq)
+	log.Printf("Modifying template %q to create %q...", old.GetName(), newTemplateName)
+	_, err := c.Insert(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	getReq := &computepb.GetInstanceTemplateRequest{
+		InstanceTemplate: newTemplateName,
+		Project:          *project,
+	}
+	return c.Get(ctx, getReq)
+}
+
+func getIGMs(ctx context.Context, c *compute.InstanceGroupManagersClient) ([]*computepb.InstanceGroupManager, error) {
+	igms := make([]*computepb.InstanceGroupManager, 0)
+	it := c.AggregatedList(ctx, &computepb.AggregatedListInstanceGroupManagersRequest{
+		Project: *project,
+	})
+	for {
+		rsp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		for _, igm := range rsp.Value.GetInstanceGroupManagers() {
+			shortZone := path.Base(igm.GetZone())
+			if !strings.HasPrefix(shortZone, *region) {
+				continue
+			}
+			igms = append(igms, igm)
+		}
+	}
+	return igms, nil
+}
+
+func createIGMWithTemplate(ctx context.Context, c *compute.InstanceGroupManagersClient, name, shortZone, modifiedTemplateURL string) error {
+	req := &computepb.InsertInstanceGroupManagerRequest{
+		Project: *project,
+		InstanceGroupManagerResource: &computepb.InstanceGroupManager{
+			Name:             proto.String(name),
+			InstanceTemplate: proto.String(modifiedTemplateURL),
+			TargetSize:       proto.Int32(int32(1)),
+		},
+		Zone: shortZone,
+	}
+
+	if !*apply {
+		log.Printf("Would create InstanceGroupManager (%+v)", req)
+		return nil
+	}
+	log.Printf("Creating InstanceGroupManager (%+v)...", req)
+	op, err := c.Insert(ctx, req)
 	if err != nil {
 		return err
 	}
-	return nil
+	return op.Wait(ctx)
 }
 
-func getTemplateURL(ctx context.Context, c *compute.InstanceTemplatesClient, templateName string) (string, error) {
-	req := &computepb.GetInstanceTemplateRequest{
-		InstanceTemplate: templateName,
-		Project:          *project,
+func deleteIGM(ctx context.Context, c *compute.InstanceGroupManagersClient, igm *computepb.InstanceGroupManager) error {
+	req := &computepb.DeleteInstanceGroupManagerRequest{
+		Project:              *project,
+		InstanceGroupManager: igm.GetName(),
+		Zone:                 path.Base(igm.GetZone()),
 	}
-	rsp, err := c.Get(ctx, req)
+
+	if !*apply {
+		log.Printf("Would delete InstanceGroupManager (%+v)", req)
+		return nil
+	}
+	log.Printf("Deleting InstanceGroupManager (%+v)...", req)
+	op, err := c.Delete(ctx, req)
 	if err != nil {
-		return "", err
+		return err
 	}
-	return rsp.GetSelfLink(), nil
+	return op.Wait(ctx)
 }
 
 func main() {
 	flag.Parse()
 
+	if !*apply {
+		defer log.Printf("Run this script again with --apply=true to execute these changes!")
+	}
+
```

Comment:
can you bail the script early if `*region == ""` or `*cluster==""`?

Reply:
Done.